### PR TITLE
Fix for Issue #12

### PR DESCRIPTION
**Pull request by jeramyRR**
Multiple Connections getting blocked at prompt for na...

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -89,7 +81,7 @@ func (s *Server) processIncoming() {
 		case msg := <-s.incoming:
 			s.SendToClients(fmt.Sprintf("%s: %s\n", msg.client.NickName, msg.msg))
 		case newConn := <-s.newConnections:
-			s.connect(newConn)
+			go s.connect(newConn)
```
Comment (ksolo) [inline server.go]: What does the go keyword do in this case?

**Comment by jeramyRR** [comment]
Go was made with concurrency as a first class citizen, and the go keyword is one of the ways to start a go routine.  Go routines can be thought of as a light weight thread, but don't necessarily have to be.  Go is smart enough to figure out the best way to run concurrent code.  It may start a new thread, run it parallel on another core, or blend it in with another running thread.  It sounds very similar to the way Elixir's concurrency works.

In this case I'm passing off the connection handling to a go routine so that the connection loop doesn't get blocked and can create other client objects as others connect.  Before, it was blocking when a person entered the prompt for name state.

This blog has a bunch of great resources on concurrency in Go, and the conference video with Rob Pike is awesome.
https://blog.golang.org/concurrency-is-not-parallelism

**Comment by ksolo** [comment]
@jeramyRR Thanks for the response!

**Comment by ksolo** [comment]
:+1:  Merging...

**Pull request merged**